Grid calendar

antoine-de opened the pull request:
Computation of the calendar association to a vj.

Stop schedule computation for calendar.

We lack for the moment the exception computation and some more test on jormungandr.

antoine-de committed:
buildbot...

antoine-de committed:
Kraken: Thermometer cleaning

remove unused data ref from thermometer + better const handling

antoine-de committed:
Protobuf: add query param in schedule

new param: calendar_id

azime committed:
kraken : Calendar2validitypattern

azime committed:
kraken : serialize AssociatedCalendar

azime committed:
kraken : remplace associated_calendar per associated_calendar*

azime committed:
kraken : refactoring add function "get_similar_validity_appetrn"

azime committed:
kreken : add function "build_associated_calendar'

azime committed:
kraken : add tests "associated_calendar"

antoine-de committed:
Kraken: associated calendar computation

antoine-de committed:
changes in departure board api

antoine-de committed:
kraken: working en calendars association

antoine-de committed:
kraken: change from one associated cal to many

the calendars associated to a vp are the calendar that match the
validity pattern of the vj for the days the calendar is active.

antoine-de committed:
Kraken: corrections on calendar association

we could associate the vehicle journey on any calendar, now we only do
that with calendars associated with the line of the vj

azime committed:
ed : rename grid calendar files

antoine-de committed:
Kraken: change to time time for calendar handling

departure board can now handle an optional calendar id.

If provided filter the stop time not on the date of the date time given
in parameters but on the fact that the vehicle journey of the stop time
is linked to the calender

azime committed:
kraken : add validity_pattern in calendar

antoine-de committed:
Jormungandr: make stop_schedule work

antoine-de committed:
Kraken: Add error if calendar does not exists

azime committed:
ed : calcul validity_pattern of calendars

azime committed:
ed : complete data

azime committed:
kraken : format datetime

azime committed:
jormungandr : add type exception_type

azime committed:
jormungandr : add tests calendars

azime committed:
jormungandr : add tests calendars

azime committed:
kraken : format date in calendars

azime committed:
ed : format date

antoine-de committed:
Kraken: work on associated calendars

azime committed:
kraken : add tests

azime committed:
jormungandr : tests

antoine-de committed:
Kraken: calendar association exception computation

azime committed:
ed : add tests for associated calendars

antoine-de committed:
Kraken: correction bug on weekdays starting on sunday

antoine-de committed:
Kraken: add unit test for departure_board

and it works!

antoine-de committed:
Kraken: date format change on departure board

if called with a calendar, only output the hour and not the date anymore

antoine-de committed:
Kraken: Add beginning of UT for exceptions in calendar association

l-vincent-l commented on source/routing/best_stoptime.h:
```diff
@@ -10,7 +10,8 @@
 earliest_stop_time(const type::JourneyPatternPoint* jpp,
               const DateTime dt,
-              const type::Data &data, bool disruption_active, bool reconstructing_path = false,
+              const type::Data &data, bool disruption_active, bool reconstructing_path,
```
je veux bien que tu remettes la valeur par défaut du bool reconstructing_path

antoine-de replied:
non.

l-vincent-l commented on source/type/pb_converter.cpp:
```diff
@@ -950,13 +958,13 @@ void fill_pb_object(const nt::Calendar* cal, const nt::Data&,
     for (const auto& excep: cal->exceptions) {
         auto pb_ex = pb_cal->add_exceptions();
-        pb_ex->set_date(boost::gregorian::to_iso_extended_string(excep.date));
+        pb_ex->set_date(boost::gregorian::to_iso_string(excep.date));
```
du coup tu vas te retrouver avec une date du type YYYYMMDDTHHmmSS non ?
Je crois que dans l'interface c'est YYYYMMDDT

antoine-de replied:
c'est boost::gregorian donc c'est juste une date ce qui donne a priori
YYYYMMDDT

On Thu, Mar 6, 2014 at 10:08 AM, l-vincent-l notifications@github.comwrote:

> In source/type/pb_converter.cpp:
> 
> > ```
> >  }
> > 
> >  for (const auto& excep: cal->exceptions) {
> >      auto pb_ex = pb_cal->add_exceptions();
> > ```
> > -        pb_ex->set_date(boost::gregorian::to_iso_extended_string(excep.date));
> > -        pb_ex->set_date(boost::gregorian::to_iso_string(excep.date));
> 
> du coup tu vas te retrouver avec une date du type YYYYMMDDTHHmmSS non ?
> Je crois que dans l'interface c'est YYYYMMDDT
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/CanalTP/kraken/pull/172/files#r10335443
> .

antoine-de committed:
Kraken: cleaning

add log
change time period to date period for calendar
restrict the calendar to the production period

kinnou02 commented on source/type/data.h:
```diff
@@ -149,10 +153,28 @@ class Data : boost::noncopyable{
     ValidityPattern* get_or_create_validity_pattern(ValidityPattern* ref_validity_pattern, const uint32_t time);
-
+    /** Get similar validitypattern **/
+    ValidityPattern* get_similar_validity_appetrn(ValidityPattern* vp) const;
```
typo

kinnou02 commented on source/ed/connectors/fusio_parser.cpp:
```diff
@@ -838,10 +838,10 @@ void FusioParser::parse_files(Data& data) {
-    parse<grid_calendar::ExceptionDatesFusioHandler>(data, "hexception_dates.txt");
-    parse<grid_calendar::CalendarLineFusioHandler>(data, "hrel_calendar_line.txt");
+    parse<grid_calendar::GridCalendarFusioHandler>(data, "grid_calendars.txt");
+    parse<grid_calendar::GridCalPeriodFusioHandler>(data, "grid_periods.txt");
```
ya peut etre pas besoin de repéter gridCal partout si c'est deja dans le namespace

kinnou02 commented on source/type/data.cpp:
```diff
@@ -137,28 +137,141 @@ void Data::build_raptor() {
-        } else {
-            ValidityPattern* tmp_vp = new ValidityPattern(ref_validity_pattern->beginning_date, tmp_days.to_string());
+        ValidityPattern vp(ref_validity_pattern->beginning_date,ref_validity_pattern->str());
+        vp.days <<= 1;
```
et ça fait quoi?

kinnou02 commented on source/type/data.cpp:
```diff
@@ -137,28 +137,141 @@ void Data::build_raptor() {
+list_cal_bitset find_matching_calendar(const Data& data, const VehicleJourney* vehicle_journey, double relative_threshold) {
+    list_cal_bitset res;
+    //for the moment we keep lot's of trace, but they will be removed after a while
+    auto log = log4cplus::Logger::getInstance("log");
```
autant donner un nom intelligent au logger, ça pourras toujours nous servir

kinnou02 commented on source/type/data.cpp:
```diff
@@ -137,28 +137,141 @@ void Data::build_raptor() {
+        //we associate the calendar to the vj if the diff are below a relative threshold
+        //compared to the number of active days in the calendar
+        size_t threshold = std::round(relative_threshold * calendar->validity_pattern.days.count());
+        LOG4CPLUS_DEBUG(log, "**** diff: " << nb_diff << " and threshold: " << threshold << (nb_diff <= threshold ? ", we keep it!!":""));
```
plutot trace non?

kinnou02 commented on source/type/data.cpp:
```diff
@@ -137,28 +137,141 @@ void Data::build_raptor() {
+        LOG4CPLUS_DEBUG(log, "the vj " << vehicle_journey->uri << " has been attached to " << cal_uri.str());
+    }
+
+    LOG4CPLUS_INFO(log, nb_matched << " vehicle journeys have been matched to at least one calendar");
```
attention, info c'est le nuveau de log par défaut quand on déploie

kinnou02 commented on source/type/type.cpp:
```diff
@@ -77,14 +78,15 @@ bool VehicleJourney::has_landing() const{
-        std::cerr << "La date est avant le début de période(" << beginning_date << ")" << std::endl;
+bool ValidityPattern::is_valid(int day) const {
+    if(day < 0) {
+        LOG4CPLUS_WARN(log4cplus::Logger::getInstance("log"), "Validity pattern not valid, the day "
```
Warn? c'est agressif quand meme

antoine-de committed:
Kraken: changes after review

kinnou02 commented on source/type/type.cpp:
```diff
@@ -77,14 +78,15 @@ bool VehicleJourney::has_landing() const{
     }
-    else if(duration > 366){
-        std::cerr << "La date dépasse la fin de période " << duration << " " << std::endl;
+    if(day > 366) {
```
c'est trés en dur quand meme

kinnou02 commented on source/type/type.h:
```diff
@@ -865,6 +859,33 @@ struct StopTime : public Nameable {
 };
 
+/**
+ * @brief The Calendar struct
```
merci captain obvious
